src/interpreter.rs

```diff
@@ -36,6 +36,33 @@ macro_rules! translate_memory_access {
     };
 }
 
+macro_rules! translate_memory_load {
```

Comment:
All this code duplication can be avoided by specializing the macro for `AccessType` and `T`, then map that to a `$method:ident` so you can call `$self.vm.env.memory_mapping.$method` or something.

Reply:
I've done this, but imo it was better before. It's about the same code length except now there's ugly macro matching trickery.